From 3c1f50a85eabdf185ab93e6dabdb8d8e5ebd79f3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Thu, 29 Jul 2021 08:44:36 +0200
Subject: [PATCH] Move D-Bus interface files to dedicated directory

Place the XML files in data/dbus-interfaces.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2126>
---
 data/dbus-interfaces/meson.build                     |  1 +
 .../org.freedesktop.RealtimeKit1.xml                 |  0
 .../dbus-interfaces}/org.freedesktop.login1.xml      |  0
 .../org.gnome.Mutter.DisplayConfig.xml               |  0
 .../org.gnome.Mutter.IdleMonitor.xml                 |  0
 .../org.gnome.Mutter.RemoteDesktop.xml               |  0
 .../dbus-interfaces}/org.gnome.Mutter.ScreenCast.xml |  0
 data/meson.build                                     |  2 ++
 src/meson.build                                      | 12 ++++++------
 9 files changed, 9 insertions(+), 6 deletions(-)
 create mode 100644 data/dbus-interfaces/meson.build
 rename {src => data/dbus-interfaces}/org.freedesktop.RealtimeKit1.xml (100%)
 rename {src => data/dbus-interfaces}/org.freedesktop.login1.xml (100%)
 rename {src => data/dbus-interfaces}/org.gnome.Mutter.DisplayConfig.xml (100%)
 rename {src => data/dbus-interfaces}/org.gnome.Mutter.IdleMonitor.xml (100%)
 rename {src => data/dbus-interfaces}/org.gnome.Mutter.RemoteDesktop.xml (100%)
 rename {src => data/dbus-interfaces}/org.gnome.Mutter.ScreenCast.xml (100%)

diff --git a/data/dbus-interfaces/meson.build b/data/dbus-interfaces/meson.build
new file mode 100644
index 00000000000..0cdb8441e93
--- /dev/null
+++ b/data/dbus-interfaces/meson.build
@@ -0,0 +1 @@
+dbus_interfaces_dir = meson.current_source_dir()
diff --git a/src/org.freedesktop.RealtimeKit1.xml b/data/dbus-interfaces/org.freedesktop.RealtimeKit1.xml
similarity index 100%
rename from src/org.freedesktop.RealtimeKit1.xml
rename to data/dbus-interfaces/org.freedesktop.RealtimeKit1.xml
diff --git a/src/org.freedesktop.login1.xml b/data/dbus-interfaces/org.freedesktop.login1.xml
similarity index 100%
rename from src/org.freedesktop.login1.xml
rename to data/dbus-interfaces/org.freedesktop.login1.xml
diff --git a/src/org.gnome.Mutter.DisplayConfig.xml b/data/dbus-interfaces/org.gnome.Mutter.DisplayConfig.xml
similarity index 100%
rename from src/org.gnome.Mutter.DisplayConfig.xml
rename to data/dbus-interfaces/org.gnome.Mutter.DisplayConfig.xml
diff --git a/src/org.gnome.Mutter.IdleMonitor.xml b/data/dbus-interfaces/org.gnome.Mutter.IdleMonitor.xml
similarity index 100%
rename from src/org.gnome.Mutter.IdleMonitor.xml
rename to data/dbus-interfaces/org.gnome.Mutter.IdleMonitor.xml
diff --git a/src/org.gnome.Mutter.RemoteDesktop.xml b/data/dbus-interfaces/org.gnome.Mutter.RemoteDesktop.xml
similarity index 100%
rename from src/org.gnome.Mutter.RemoteDesktop.xml
rename to data/dbus-interfaces/org.gnome.Mutter.RemoteDesktop.xml
diff --git a/src/org.gnome.Mutter.ScreenCast.xml b/data/dbus-interfaces/org.gnome.Mutter.ScreenCast.xml
similarity index 100%
rename from src/org.gnome.Mutter.ScreenCast.xml
rename to data/dbus-interfaces/org.gnome.Mutter.ScreenCast.xml
diff --git a/data/meson.build b/data/meson.build
index b55ded90896..ae983219980 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -64,3 +64,5 @@ if have_libgudev
     install_dir: join_paths(udev_dir, 'rules.d'),
   )
 endif
+
+subdir('dbus-interfaces')
diff --git a/src/meson.build b/src/meson.build
index 7dba903d732..6e043d5e862 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -824,14 +824,14 @@ if mutter_private_enum_sources.length() > 0
 endif
 
 dbus_display_config_built_sources = gnome.gdbus_codegen('meta-dbus-display-config',
-    'org.gnome.Mutter.DisplayConfig.xml',
+    join_paths(dbus_interfaces_dir, 'org.gnome.Mutter.DisplayConfig.xml'),
     interface_prefix: 'org.gnome.Mutter.',
     namespace: 'MetaDBus',
   )
 mutter_built_sources += dbus_display_config_built_sources
 
 dbus_idle_monitor_built_sources = gnome.gdbus_codegen('meta-dbus-idle-monitor',
-    'org.gnome.Mutter.IdleMonitor.xml',
+    join_paths(dbus_interfaces_dir, 'org.gnome.Mutter.IdleMonitor.xml'),
     interface_prefix: 'org.gnome.Mutter.',
     namespace: 'MetaDBus',
     object_manager: true,
@@ -870,7 +870,7 @@ if have_native_backend
   mutter_built_sources += default_modes_h
 
   dbus_login1_built_sources = gnome.gdbus_codegen('meta-dbus-login1',
-    'org.freedesktop.login1.xml',
+    join_paths(dbus_interfaces_dir, 'org.freedesktop.login1.xml'),
     interface_prefix: 'org.freedesktop.login1.',
     namespace: 'MetaDbusLogin1',
   )
@@ -879,14 +879,14 @@ endif
 
 if have_remote_desktop
   dbus_remote_desktop_built_sources = gnome.gdbus_codegen('meta-dbus-remote-desktop',
-      'org.gnome.Mutter.RemoteDesktop.xml',
+      join_paths(dbus_interfaces_dir, 'org.gnome.Mutter.RemoteDesktop.xml'),
       interface_prefix: 'org.gnome.Mutter.',
       namespace: 'MetaDBus',
     )
   mutter_built_sources += dbus_remote_desktop_built_sources
 
   dbus_screen_cast_built_sources = gnome.gdbus_codegen('meta-dbus-screen-cast',
-      'org.gnome.Mutter.ScreenCast.xml',
+      join_paths(dbus_interfaces_dir, 'org.gnome.Mutter.ScreenCast.xml'),
       interface_prefix: 'org.gnome.Mutter.',
       namespace: 'MetaDBus',
     )
@@ -894,7 +894,7 @@ if have_remote_desktop
 endif
 
 dbus_rtkit_built_sources = gnome.gdbus_codegen('meta-dbus-rtkit1',
-    'org.freedesktop.RealtimeKit1.xml',
+    join_paths(dbus_interfaces_dir, 'org.freedesktop.RealtimeKit1.xml'),
     interface_prefix: 'org.freedesktop.',
     namespace: 'MetaDbus',
 )
-- 
GitLab
